src/main/java/input_feed/distributed/TextToDistributedR1CS.java

```diff
@@ -0,0 +1,156 @@
+                distributedCombination(config, filePath + ".c", negateCMatrix);
+        config.endRuntime("Distribute Constraints");
+
+        config.beginRuntime("Count combinations");
+
+        config.beginLog("Count A");
+        combinationA.count();
+        config.endLog("Count A");
+
+        config.beginLog("Count B");
+        combinationB.count();
+        config.endLog("Count B");
+
+        config.beginLog("Count C");
+        combinationC.count();
+        config.endLog("Count C");
+
+        config.endRuntime("Count combinations");
+
+        final R1CSConstraintsRDD<FieldT> constraints = new R1CSConstraintsRDD<>(
+                combinationA,
+                combinationB,
+                combinationC,
+                numConstraints
+        );
+
+        return new R1CSRelationRDD<>(constraints, numInputs, numAuxiliary);
+    }
+
+    public Tuple2<Assignment<FieldT>, JavaPairRDD<Long, FieldT>> loadWitness(Configuration config) {
+
+        config.beginRuntime("Distribute Witness");
+        config.beginLog("Distribute witness");
+
+        FieldT field = fieldParameters;
+        String path = filePath;
+        if (useFileSlash) { path = "file://" + path; }
+
+        final int numAux = numAuxiliary;
+        JavaPairRDD<Long, FieldT> auxAssignment =
+                config.sparkContext().textFile(path + ".aux").zipWithIndex().flatMapToPair(
+                        line -> Collections.singleton(new Tuple2<>(line._2, field.construct(line._1))).iterator()
+                ).persist(config.storageLevel());
+
+        JavaPairRDD<Long, FieldT> primaryAssignment =
+                config.sparkContext().textFile(path + ".public").zipWithIndex().flatMapToPair(
+                        line -> Collections.singleton(new Tuple2<>(numAux + line._2, field.construct(line._1))).iterator()
+        ).persist(config.storageLevel());
+
+        JavaPairRDD<Long, FieldT> fullAssignment = auxAssignment.union(primaryAssignment);
+
+        config.beginLog("Count assignment");
+        fullAssignment.count();
+        config.endLog("Count assignment");
+
+        config.endLog("Distribute witness");
+        config.endRuntime("Distribute Witness");
+
+        config.beginLog("Serialize Primary");
+        config.beginRuntime("Serialize Primary");
+        final Assignment<FieldT> primary = new Assignment<>();
+        for (long i=0; i < numInputs; i++){
+            List<FieldT> element = fullAssignment.lookup(i);
+            assert(element.size() == 1);
+            primary.add(element.get(0));
+        }
+        config.endLog("Serialize Primary");
+        config.endRuntime("Serialize Primary");
+
+        return new Tuple2<>(primary, fullAssignment);
+    }
+
+    private JavaPairRDD<Long, LinearTerm<FieldT>>
+    distributedCombination(Configuration config, String fileName, boolean negate) {
+        FieldT field = fieldParameters;
+        String path = fileName;
+        if (useFileSlash) { path = "file://" + path; }
```

Comment:
Wouldn't it be simpler to just add the file:// to the `String fileName` argument in the cases where it is needed, rather than adding this information to the class?

Reply:
Not in cases where I want to run this stuff on my local machine. The file slash is only necessary on the AWS instance. It was a pain to keep adding them an removing them when moving between local and remote.